fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/scalar/BinaryPredicateOperator.java

```diff
@@ -154,4 +154,28 @@ public int hashCode() {
     public boolean isNullable() {
         return !this.type.equals(BinaryType.EQ_FOR_NULL) && super.isNullable();
     }
+
+    public static BinaryPredicateOperator eq(ScalarOperator... arguments) {
```

Comment:
BinaryPredicateOperator Since it is limited to a binary operator here, why do we set a variable length parameter? Is it better to change to fixed parameters instead of relying on Preconditions.checkState(arguments.size() == 2);
Throws an Unknow Error to the user at runtime.

Reply:
You are right